packages/common_client/lib/src/data_sources/data_source_event_handler.dart

```diff
@@ -104,7 +104,6 @@ final class DataSourceEventHandler {
       _statusManager.setValid();
       return MessageStatus.messageHandled;
     } catch (err) {
-      // TODO sc-215195: fix contract test that is failing here due to empty flag content
```

Comment:
This was not why contract tests failed here. It was because the data was invalid. For a client SDK an empty data set is `{}` for a server is it `{"flags":{},"segments":{}}`, which is an invalid payload. This has been fixed in the contract tests.